website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,540 @@
+
+Depending on the use case, you might want to use a date field, or a text field, or a select field. In the UIKit library we provide a variety of components to cover different use cases. For more information about field and input components check their related packages in the [commercetools/ui-kit](https://github.com/commercetools/ui-kit) repository.
+
+<Info>
+
+All UIKit field components have a related input component, for example `<TextField>` -> `<TextInput>`.
+
+</Info>
+
+## Accessibility support
+
+All field and input components have built-in support for accessibility features such as aria labels, keyboard navigation and focus, error messaging.
+
+<Info>
+
+Testing these field components with [React Testing Library](https://github.com/testing-library/react-testing-library) can be easily done using selectors such as `*ByLabelText`, `*ByRole`, etc.
+
+</Info>
+
+# Form validation
+
+One of the important parts of forms is validation. It's important to validate things like required fields, or to check that the value is a valid URL, or even perform asynchronous checks against an API.
+
+For that, Formik allows you to implement a [`validate` function](https://formik.org/docs/api/formik#validate-values-values--formikerrorsvalues--promiseany) that should return an object of errors.
+
+```ts
+type FieldErrors = Record<string, boolean>;
+type FormikErrors = Record<string, FieldErrors>;
+
+declare const validate = (values: FormikValues) => FormikErrors;
+```
+
+Specifically, the `FormikErrors` object should contain properties as the name of the form fields with values being objects with error keys. Let's take a look at an example:
+
+```js title="validate.js"
+import TextInput from '@commercetools-uikit/text-input';
+
+const validate = (values) => {
+  const errors = {};
+
+  if (TextInput.isEmpty(values.key)) {
+    errors.key = { missing: true };
+  }
+
+  return errors;
+}
+```
+
+In this example we are validating that the `key` field has a required value. If the value is empty, we assign to the `errors.key` property an error object with the error key `missing` set to `true`.
+
+In the field component you need to make sure to assign the `errors` and `touched` props which will make the field component render an error message, in case one was returned from the `validate` function.
+
+```js highlightLines="6-7"
+<TextField
+  name="key"
+  title="Key"
+  isRequired={true}
+  value={formik.values.key}
+  errors={formik.errors.key}
+  touched={formik.touched.key}
+  onChange={formik.handleChange}
+  onBlur={formik.handleBlur}
+/>
+```
+
+<Info>
+
+Error messages are only shown when the `touched` value for the specific field is set to `true`. This happens whenever the user stops interacting with a field (loses `focus`). The point here is that as long as the user is interacting with a field for the first time there is no need to show a validation.
```

Comment:
```suggestion
Error messages are only shown when the `touched` value for the specific field is set to `true`. This happens whenever the user stops interacting with a field and moves to another so that the prior field loses `focus`. The point here is that as long as the user is interacting with a field for the first time there is no need to show a validation.
```

Reply:
That's not always true. Clicking outside of the input also loses focus.

I think I'll keep it like this for now, if you don't mind.